python_modules/dagster/dagster/core/asset_defs/decorators.py

```diff
@@ -278,6 +280,8 @@ def __call__(self, fn: Callable) -> AssetsDefinition:
             for key in self.required_resource_keys:
                 required_resource_keys.add(key)
             for key in self.resource_defs.keys():
+                if isinstance(self.resource_defs[key], IOManagerDefinition):
```

Comment:
What's the rationale for this check? Also, best to avoid `continue` when not needed. Could just do `if isinstance(self.resource_defs[key], IOManagerDefinition):`?

Reply:
This check is to ensure we don't accidentally make io managers available to the body of an op via the resources object, which we don't do elsewhere.